src/engine/board.ts

```diff
@@ -49,13 +51,36 @@ export default class Board {
         return board;
     }
 
-    public squaresReachableInDirection(square: Square, direction : Offset){
+    private inBounds(square:Square) {
+        return square.row < gameSettings.BOARD_SIZE && square.col < gameSettings.BOARD_SIZE && square.row >= 0 && square.col >= 0
+    }
+
+    public offsetSquareAndCheckBounds(square:Square, offset:Offset):Square | null{
```

Comment:
A function with `And` in the name is inevitably doing two things, so should probably be two separate functions!

Externally to this class, this is only used within the pawn-related code, and I think it should be possible to refactor that code to avoid calling this method.

I think `inBounds` is the more natural method to make public here; it allows us to check a property of an object that we have against the board. `offsetSquareAndCheckBounds` is actually _envying_ the features of `square`, because it's calling `square.offset`.